protocol/communities/manager.go

```diff
@@ -2408,6 +2410,18 @@ func (m *Manager) HandleCommunityCancelRequestToJoin(signer *ecdsa.PublicKey, re
 		return nil, err
 	}
 
+	if community.HasMember(signer) && request.Clock > previousRequestToJoin.Clock {
```

Comment:
I think it makes sense to check the clock a bit earlier, before we mark request to join as canceled.
```
if request.Clock <= previousRequestToJoin.Clock {
    return nil, ErrInvalidClock
}
```